src/mcu.cpp

```diff
@@ -1,12 +1,12 @@
 /**
- * @file mcu.c
+ * @file mcu.cpp
  *
  * @brief MCU related
  */
 
 #include <stdint.h>
 
-#include "mcu.h"
+#include "mcu.hpp"
 
 #include "gpio.h"
 #include "main.h"
```

Comment:
```suggestion
#include "main.h"

namespace hal {
```
Como aqui a gente só tá definindo a classe que tá dentro do namespace, acho bem suave de fazer isso daqui.